plugin-taskbar/lxqttaskbutton.cpp

```diff
@@ -173,16 +176,21 @@ void LXQtTaskButton::refreshIconGeometry(QRect const & geom)
                     NET::WMIconGeometry,
                     NET::Properties2());
     NETRect const curr = info.iconGeometry();
-    if (curr.pos.x != geom.x() || curr.pos.y != geom.y()
-            || curr.size.width != geom.width() || curr.size.height != geom.height())
-    {
-        NETRect nrect;
-        nrect.pos.x = geom.x();
-        nrect.pos.y = geom.y();
-        nrect.size.height = geom.height();
-        nrect.size.width = geom.width();
-        info.setIconGeometry(nrect);
-    }
+
+    // see kwindowsystem -> NETWinInfo::setIconGeometry for the scale factor
+    const qreal scaleFactor = qApp->devicePixelRatio();
```

Comment:
Don't you want to use the widget's devicePixelRatio()?

Reply:
No, because `NETWinInfo::setIconGeometry` uses app's device ratio.